src/daemon/backends/connman_manager.h

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2019 Luxoft Sweden AB
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#ifndef CONNECTIVITY_MANAGER_DAEMON_BACKENDS_CONNMAN_MANAGER_H
+#define CONNECTIVITY_MANAGER_DAEMON_BACKENDS_CONNMAN_MANAGER_H
+
+#include <giomm.h>
+#include <glibmm.h>
+#include <sigc++/sigc++.h>
+
+#include <tuple>
+#include <vector>
+
+#include "daemon/backends/connman_agent.h"
+#include "daemon/backends/connman_service.h"
+#include "daemon/backends/connman_technology.h"
+#include "generated/dbus/connman_proxy.h"
+
+namespace ConnectivityManager::Daemon
+{
+    // Helper class for ConnMan manager. See doc/manager-api.txt in the ConnMan repo.
+    //
+    // Encapsulates asynchronous creation of D-Bus proxy, listens for manager changes in ConnMan and
+    // delegates creation of technologies and services etc.
+    class ConnManManager : public sigc::trackable
+    {
+    public:
+        class Listener;
+
+        explicit ConnManManager(Listener &listener);
+
+        Glib::RefPtr<Gio::DBus::Connection> dbus_connection() const
+        {
+            return proxy_->dbusProxy()->get_connection();
+        }
+
+        void register_agent(const ConnManAgent &agent);
+
+    private:
+        using Proxy = net::connman::ManagerProxy;
+
+        using ServicePropertiesArray =
+            std::vector<std::tuple<Glib::DBusObjectPathString, ConnManService::PropertyMap>>;
+        using TechnologyPropertiesArray =
+            std::vector<std::tuple<Glib::DBusObjectPathString, ConnManTechnology::PropertyMap>>;
+
+        void proxy_create_finish(const Glib::RefPtr<Gio::AsyncResult> &result);
+
+        void name_owner_changed() const;
+
+        void get_technologies_finish(const Glib::RefPtr<Gio::AsyncResult> &result) const;
```

Comment:
What is the `get` part of this function? There are no return values and no output variables...
Also does finish mean that it is blocking or something else?

Reply:
The `get_` part comes from the fact that it is the slot/method called when the asynchronous D-Bus method "GetTechnologies" is finished. You will see the connection if you look at where it is used in `connman_manager.cpp`.